packages/typespec-azure-resource-manager/lib/operations.tsp

```diff
@@ -108,9 +108,32 @@ op ArmResourceRead<
   ErrorResponse
 >;
 
+/**
+ * A resource HEAD operation
+ * @template Resource the resource being checked
+ * @template BaseParameters Optional. Allows overriding the operation parameters
+ * @template Parameters Optional. Additional parameters after the path parameters
+ * @template Response Optional. The success response for the read operation
+ * @template Error Optional. The error response, if non-standard.
+ */
+@autoRoute
+@doc("Check a {name} via HEAD operation", Resource)
+@head
+op ArmResourceHead<
+  Resource extends ArmResourceBase,
+  BaseParameters = DefaultBaseParameters<Resource>,
+  Parameters extends {} = {},
+  Response extends {} = OkResponse,
```

Comment:
This should return one of the ARM responses, either with an empty body, or no content.  (i.e. ArmResponse).  Also, don't we specifically want to describe the 404 in this case, since this should not be treated as a standard error?

Reply:
Looking through existing specs, especially this: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/stable/2023-07-01/resources.json#L2995-L3069

I think the two default responses should be NoContent and NotFound (204, 404)

Reply:
Returning a 204 would be wrong/violating RFC9110 (unless the corresponding GET request would have returned 204). So I really hope that is not what ARM does...

Reply:
Well, following the link @markcowl provided above. And now I'm sad.

Reply:
ARM indeed picked `204` over `200` because of empty body. As this is only limited in ARM lib, Azure.Core or TCGC could still add support as add-on to existing GET as you have suggested. As for impl, while it is easy to slap on a `@EnableHead()` on any GET operation there, it would tricky for  customizations such as ResponseCode and potentially additional headers.